### PR TITLE
Change to remote ejb server implementation

### PR DESCRIPTION
**Pull request by tadamski**
This commit is a part of integrating RemoteSever with ejb3 subsystem.

### CODE REVIEW

File: src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java [hunk 1]
```diff
@@ -288,10 +290,10 @@ public void processInvocation(final EJBReceiverInvocationContext receiverContext
                     }
 
                     // write sec context
-                    marshaller.writeInt(peerIdentityId);
+                    marshaller.writeShort(peerIdentityId);
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java]: These have to be 4 byte integers.

File: src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java [hunk 2]
```diff
@@ -539,7 +546,7 @@ public void writeResponse(final Object response) {
             }
         }
 
-        public void writeException(final Exception response) {
+        public void writeThrowable(final Throwable response) {
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java]: We don't ever want to write Errors or anything like that; generally the EJB protocol and spec deal in Exceptions only.  Was there some reason this had to be widened?

Comment (tadamski) [inline src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java]: Current org.jboss.as.ejb3.remote.protocol.versionone.SessionOpenRequestHandler catches Throwable during session id generation so I assumed that some Errors may be thrown during session id generation.

File: src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java (hunk 1 above)
Comment (tadamski) [inline src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java]: Fixed on client and server side.

File: src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java [hunk 3]
```diff
@@ -308,14 +303,20 @@ void handleInvocationRequest(final int invId, final InputStream input) throws IO
                 int transactionId = unmarshaller.readUnsignedShort();
                 identity = identityId == 0 ? connection.getLocalIdentity() : connection.getLocalIdentity(identityId);
                 transaction = null; // todo: look up transaction by transactionId
+                locator = unmarshaller.readObject(EJBLocator.class);
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java]: Moving these methods changes the protocol (see protocol.txt).  I think the app/mod name should come first, as described in that doc. Did you encounter a reason to change it?

File: src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java [hunk 4]
```diff
@@ -237,14 +238,15 @@ public void processInvocation(final EJBReceiverInvocationContext receiverContext
         invocationContext.putAttachment(INV_KEY, invocation);
         final EJBLocator<?> locator = invocationContext.getLocator();
         final int peerIdentityId;
-        if (version >= 3) try {
-            peerIdentityId = channel.getConnection().getPeerIdentityId();
-        } catch (AuthenticationException e) {
-            receiverContext.resultReady(new EJBReceiverInvocationContext.ResultProducer.Failed(e));
-            return;
-        } else {
+        //TODO Elytron
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java]: This should be functional in the latest Remoting/Elytron versions.  Did you encounter a problem?

File: src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java [hunk 5]
```diff
@@ -416,7 +416,7 @@ private void writeFailedResponse(final int invId, final Exception e) {
             this.transaction = transaction;
         }
 
-        int getInvId() {
+        public int getInvId() {
```
Comment (tadamski) [inline src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java]: This is used in Association class implementation to trace incomming invocations on the server side.

File: src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java (hunk 4 above)
Comment (tadamski) [inline src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java]: Encountered NullPointerException - checking versions...

**Comment by tadamski** [comment]
Failure happens because I have made the version of jboss-remoting 5.0.0.Beta13 - will retest after the release.

**Comment by dmlloyd** [comment]
I've released 5.0.0.Beta13 of Remoting.  Once that's updated in the POM I will merge this.

**Comment by tadamski** [comment]
I have already made it 5.0.0.Beta13 in the PR. 
Retest this please

**Comment by dmlloyd** [comment]
Retest this please

**Comment by dmlloyd** [comment]
I manually restarted the test, looks like CI has cached the non-existence of this release.  I'll try it again.

**Pull request merged**